### DISCUSSION TITLE
dotnet nuget push not working after GitHub packages changes (10/10/22)

### DISCUSSION THREAD

**Discussion by Sergio0694** [community/community#36230] [Packages] [answered]
Posting here after https://github.com/dotnet/sdk/issues/28545 and https://github.com/NuGet/Home/issues/12161 have been closed.

---

### NuGet Product Used

dotnet.exe

### Product Version

Using 6.0.402 .NET SDK, but don't think it's version specific

### Worked before?

Yes, been working fine for months

### Impact

I'm unable to use this version (I can't push to GitHub packages at all)

### Repro Steps & Context

This is a new issue after https://github.com/dotnet/sdk/issues/28545 was closed and redirected here.

---

### Describe the bug

It seems the new "Consolidated owner and admin role" changes in GitHub packages are causing CI steps pushing to GitHub packages to fail, and I couldn't find a clear explanation on how to fix this through `dotnet nuget push`. See: https://github.com/Sergio0694/ComputeSharp/issues/403. I'm currently using this command:

```
dotnet nuget push "artifacts\*.nupkg" --source "https://nuget.pkg.github.com/ComputeSharp/index.json" --api-key ${{secrets.GITHUB_TOKEN}} --skip-duplicate
```

Which so far has always worked just fine. This broke a few days ago following those changes. Is there guidance on how to update scripts using `dotnet nuget push` to account for these recent GitHub changes? Not sure what additional params I should pass, if any.

### To Reproduce

See this CI run: https://github.com/Sergio0694/ComputeSharp/actions/runs/3242075036/jobs/5315478395.

### Exception

```
Pushing ComputeSharp.Pix.2.0.0-alpha.3242075036+6131b7f9b0d2ee95c98e5e76a5f5bafe62ef74ae.nupkg to 'https://nuget.pkg.github.com/ComputeSharp'...
  PUT https://nuget.pkg.github.com/ComputeSharp/
warn : Source owner 'ComputeSharp' does not match repo owner 'Sergio0694' in repository element.
  BadRequest https://nuget.pkg.github.com/ComputeSharp/ 380ms
error: Response status code does not indicate success: 400 (Bad Request).
```

### Further technical details

- Using the 6.0.402 .NET SDK

**Comment by ChristopherHX** [comment, marked as answer]
You have used the wrong nuget package source url:
You used
`https://nuget.pkg.github.com/ComputeSharp/index.json`, but ComputeSharp is not the owner of your ComputeSharp repository. No idea why it worked before.

You need to use https://nuget.pkg.github.com/Sergio0694/index.json, this worked for me two days ago with my own repo ( https://github.com/ChristopherHX/runner.server/actions/runs/3243655185/jobs/5318681676#step:10:1 ).

I use this in actions:
```
dotnet nuget push ./*.nupkg --source "https://nuget.pkg.github.com/${{ github.repository_owner }}/index.json" --api-key ${{ secrets.GITHUB_TOKEN }} --skip-duplicate --no-symbols
```

**Reply by Sergio0694** [reply]
Ooh I see, thank you!! Let me try this out 😄

**Reply by Sergio0694** [reply]
That worked! Thank you again! 🎉

**Comment by ossanna16** [comment]
Hey there @Sergio0694 and thank you for asking a good question! I'm glad you found our community 🙂
